shared/constants/chat2/message.js

```diff
@@ -972,7 +973,25 @@ const imageFileNameRegex = /[^/]+\.(jpg|png|gif|jpeg|bmp)$/i
 export const pathToAttachmentType = (path: string) => (imageFileNameRegex.test(path) ? 'image' : 'file')
 export const isSpecialMention = (s: string) => ['here', 'channel', 'everyone'].includes(s)
 
-export const upgradeMessage = (old: Types.Message, m: Types.Message) => {
+export const upgradeMessage = (_old: ?Types.Message, m: Types.Message, merge: boolean) => {
+  if (merge) {
+    if (_old) {
+      // $FlowIssue doens't understand mergeWith
+      return _old.mergeWith((oldVal, newVal, key) => {
+        if (key === 'mentionsAt' || key === 'reactions') {
+          return shallowEqual(oldVal, newVal) ? oldVal : newVal
```

Comment:
`mentionsAt` and `reactions` are both immutable. AIUI since the messages coming from `messagesAdd` are new records created from the thread load, these keys can only be shallow equal if old and new are both empty. `Immutable.is` might be better here

Reply:
updated, tnx!